### DISCUSSION TITLE
Setting a layer as active when added to a layergroup, hover instead of click to show marker popup.

### DISCUSSION THREAD

**Discussion by ero-qt** [Leaflet/Leaflet#8797] [Q&A] [answered]
I'm dynamically adding layers to the map simply using `myLayers.addBaseLayer(someLayer, someName)`. This works fine, but all layers added are automatically deselected, such that no map renders when the page loads, and a layer needs to be selected instead.  
Is it possible to forcibly select a layer when it's added?

---

Is it possible to have a marker's popup to appear when the marker is being hovered instead of clicked?

I couldn't find anything about either of these things in the docs.

**Comment by IvanSanchez** [comment, marked as answer]
Do something like `someLayer.addTo(map)`, in addition to what you're already doing. As noted in the `L.Control.Layers` tutorial, the layers control just reflects whether the layer is in the map or not.

---

Is it possible to have a marker's popup to appear when the marker is being hovered instead of clicked?

`marker.on('mouseover', (ev)=>ev.target.openPopup());`

**Reply by ero-qt** [reply]
> `marker.on('mouseover', (ev)=>ev.target.openPopup());`

This doesn't seem to work. The marker renders, but it doesn't do anything on hover. No error or anything.

```js
fetch('overlays.json')
  .then(response => response.json())
  .then(data => {
    for (const layer in data) {
      const group = L.layerGroup();

      for (const item in data[layer]) {
        const pos = data[layer][item];
        const marker = L.marker([pos[0], pos[1]]);

        marker.on('mouseover', (ev) => ev.target.openPopup());
        marker.addTo(group);
      }

      _layers.addOverlay(group, layer);
      group.addTo(_map);
    }
  });
```

**Reply by ero-qt** [reply]
Might help if I gave it a popup in the first place. I simply switched to `bindTooltip` though and that seems to work fine.
